### PR TITLE
fix patch_attention_mask incorrect setting which leads to the differe…

### PR DESCRIPTION
**Pull request by sywangyi**
…nce in the generated text if batch > 1

Fixes # (issue)
found the evaluate accuracy differ between batch=1 and batch>1.
if batch>1, some image would have padding pixel. and patch_attention_mask incorrect setting makes the generated result different.
- vision models: @amyeroberts, @qubvel

### CODE REVIEW

File: tests/models/idefics2/test_modeling_idefics2.py [hunk 1]
```diff
@@ -540,6 +540,41 @@ def test_integration_test_4bit(self):
...
+        model = Idefics2ForConditionalGeneration.from_pretrained(
+            "HuggingFaceM4/idefics2-8b-base",
+            load_in_4bit=True,
+        )
+
+        from datasets import load_dataset
+
+        dataset = load_dataset("nielsr/docvqa_1200_examples", split="test")
+
+        text = [f"<image>{dataset[40]['query']['en']}", f"<image>{dataset[41]['query']['en']}"]
```
Comment (amyeroberts) [inline tests/models/idefics2/test_modeling_idefics2.py:557]: Out of interest - why samples 40 and 41?

**Review by amyeroberts — APPROVED** [review]
Thanks for fixing and adding a test! 

Final step is running the slow tests for the model before merge. Could you push an empty commit with the message `[run_slow] idefics2`?

File: tests/models/idefics2/test_modeling_idefics2.py (hunk 1 above)
Comment (sywangyi) [reply tests/models/idefics2/test_modeling_idefics2.py:557]: found total 64 difference in nielsr/docvqa_1200_examples test dataset caused by such issue. just pick up sample40,41 as an example
